### PR TITLE
added router

### PR DESCRIPTION
**Pull request by ShishkunovD**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  text: String,
+  isCheck: Boolean
+})
+
+const Task = mongoose.model("tasks", taskScheme);
+
+app.use(cors());
+
+const url = 'mongodb+srv://ShishkunovD:restart987!@cluster0.8tbt5.mongodb.net/Todo-list-JS?retryWrites=true&w=majority';
+mongoose.connect(url, {useNewUrlParser: true, useUnifiedTopology: true});
```
Comment (NataliaShchelkunova) [inline app.js]: why you use {useNewUrlParser: true, useUnifiedTopology: true}?

File: app.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+app.use(bodyParser.json());
```
Comment (NataliaShchelkunova) [inline app.js]: use express

File: app.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+app.get('/allTasks', (req, res) => {
+  Task.find().then(result => {
+    res.send({data: result});
+  })
+})
+
+app.post('/createTask', (req, res) => {
+  const task = new Task(req.body);
+  task.save().then(result => {
+    res.send('Task created');
```
Comment (NataliaShchelkunova) [inline app.js]: res.send({ data: result })

File: app.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  })
+})
+
+app.delete('/deleteTask', (req, res) => {
+  const identifier = req.query.id;
+  Task.deleteOne({_id : identifier}).then(result => {
+    res.send('Task deleted');
```
Comment (NataliaShchelkunova) [inline app.js]: res.send({ data: result })

File: app.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+app.post('/createTask', (req, res) => {
+  const task = new Task(req.body);
+  task.save().then(result => {
+    res.send('Task created');
+  })
+})
+
+app.delete('/deleteTask', (req, res) => {
+  const identifier = req.query.id;
+  Task.deleteOne({_id : identifier}).then(result => {
```
Comment (NataliaShchelkunova) [inline app.js]: please, use find() in result for Task after deleteOne

File: app.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    res.send('Task deleted');
+  });
```
Comment (NataliaShchelkunova) [inline app.js]: what happens if  the user doesn't send _id?

File: app.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+})
+
+app.put('/updateTask', (req, res) => {
+  const task = req.body._id;
+  Task.findByIdAndUpdate(task, { text : req.body.text }).then(result => {
```
Comment (NataliaShchelkunova) [inline app.js]: what happens if the user doesn't send, text?

File: app.js [hunk 8]
```diff
@@ -20,56 +20,44 @@ app.use(express.json());
 
 app.get('/allTasks', (req, res) => {
   Task.find().then(result => {
-    res.send({data: result});
+    res.send({ data: result });
   })
 })
 
 app.post('/createTask', (req, res) => {
   const task = new Task(req.body);
```
Comment (usatovaDasha) [inline app.js]: if i don't send data?

File: app.js [hunk 9]
```diff
@@ -20,56 +20,44 @@ app.use(express.json());
...
   }
 })
 
 app.patch('/updateTask', (req, res) => {
   const params = req.query.id;
-  const body = req.body;
-  if(req.body.text === undefined) {
-    res.send("Пожалуйста, заполните текст")
+  const body = req.body.text;
+  if(body === undefined) {
```
Comment (usatovaDasha) [inline app.js]: only text need?

File: app.js [hunk 10]
```diff
@@ -20,56 +20,44 @@ app.use(express.json());
...
+    res.send("Пожалуйста, заполните поле текста.");
```
Comment (usatovaDasha) [inline app.js]: error status

Comment (usatovaDasha) [inline app.js]: english lang message

File: app.js [hunk 11]
```diff
@@ -20,56 +20,44 @@ app.use(express.json());
...
+  } else if(params === undefined) {
```
Comment (usatovaDasha) [inline app.js]: spaces

File: app.js [hunk 12]
```diff
@@ -26,15 +26,19 @@ app.get('/allTasks', (req, res) => {
...
+  } else {
+    task.save().then(result => {
+      res.send({ data: result });
+    })
+  }
 })
 
 app.delete('/deleteTask', (req, res) => {
   const identifier = req.query.id;
   if(identifier === undefined) {
```
Comment (usatovaDasha) [inline app.js]: spaces

File: app.js [hunk 13]
```diff
@@ -45,13 +49,11 @@ app.delete('/deleteTask', (req, res) => {
 
 app.patch('/updateTask', (req, res) => {
   const params = req.query.id;
-  const body = req.body.text;
-  if(body === undefined) {
-    res.send("Пожалуйста, заполните поле текста.");
-  } else if(params === undefined) {
-    res.send("Пожалуйста, заполните id.");
+  const body = req.body;
+  if(body.text === undefined || params === undefined || typeof body.isCheck !== "boolean") {
```
Comment (usatovaDasha) [inline app.js]: id isn't required?

File: app.js [hunk 14]
```diff
@@ -45,13 +49,11 @@ app.delete('/deleteTask', (req, res) => {
...
+    res.status(404).send("Error, all fields should be filled.");
   } else {
-    Task.findByIdAndUpdate(params, { text: body })
+    Task.findByIdAndUpdate(params, { text: body.text })
```
Comment (usatovaDasha) [inline app.js]: only text?

File: app.js [hunk 15]
```diff
@@ -25,19 +25,23 @@ app.get('/allTasks', (req, res) => {
 })
 
 app.post('/createTask', (req, res) => {
-  const task = new Task(req.body);
-  if(req.body.text === undefined || typeof req.body.isCheck !== "boolean") {
-    res.status(404).send("Error, all fields should be filled in correctly.");
+  if( req.body.text !== undefined ) {
```
Comment (usatovaDasha) [inline app.js]: if( req.body.text !== undefined  && typeof req.body.isCheck !== "boolean" ) {

File: app.js [hunk 16]
```diff
@@ -50,13 +54,20 @@ app.delete('/deleteTask', (req, res) => {
 app.patch('/updateTask', (req, res) => {
   const params = req.query.id;
   const body = req.body;
-  if(body.text === undefined || params === undefined || typeof body.isCheck !== "boolean") {
-    res.status(404).send("Error, all fields should be filled.");
+  if( params !== undefined ) {
```
Comment (usatovaDasha) [inline app.js]: if( params !== undefined && (body.text !== undefined || typeof body.isCheck === "boolean" )) {

File: app.js [hunk 17]
```diff
@@ -50,13 +54,20 @@ app.delete('/deleteTask', (req, res) => {
...
+    if( body.text !== undefined && typeof body.isCheck === "boolean" ) {
+      Task.findByIdAndUpdate(params, { text: body.text })
```
Comment (usatovaDasha) [inline app.js]: combine to one

**Pull request merged**